src/main/java/com/enderio/EnderIO.java

```diff
@@ -84,6 +85,7 @@ public EnderIO() {
         // Run datagen after registrate is finished.
         IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
         modEventBus.addListener(EventPriority.LOWEST, this::onGatherData);
+        modEventBus.addListener(SoulVialItem::onCommonSetup);
```

Comment:
Could we use annotations to subscribe this event instead, like for the other events?

Reply:
it's not as easy, as you can only have one Mod.EventBusSubscriber Annotation on a class, but the SoulVialItem needs to listen to ForgeEventBus and ModEventBus Events, which is not possible purely with Annotations. (Maybe with a static inner class, but that feels more complicated than that one call here)

Reply:
Ahh I understand, I'm torn between suggesting an inner clast and just putting it there. My only issue with putting it in here is the potential for the list of stuff in this class being registered to grow and become annoying. Do whichever you see fit, thanks for explaining it :)